modules/machineset-yaml-azure.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+// * machine_management/creating-machineset-azure.adoc
+
+[id="machineset-yaml-azure_{context}"]
+=  Sample YAML for a MachineSet Custom Resource on Azure
+
+This sample YAML defines a MachineSet that runs in the `centralus1`
+Microsoft Azure region and creates nodes that are labeled with
+`node-role.kubernetes.io/<role>: ""`
+
+In this sample, `<infrastructureID>` is the infrastructure ID label that is
+based on the cluster ID that you set when you provisioned
+the cluster, and `<role>` is the node label to add.
+
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+    machine.openshift.io/cluster-api-machine-role: <role> <2>
+    machine.openshift.io/cluster-api-machine-type: <role> <2>
+  name: <infrastructureID>-<role>-<region> <3>
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+      machine.openshift.io/cluster-api-machineset: <infrastructureID>-<role>-<region> <3>
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+        machine.openshift.io/cluster-api-machine-role: <role> <2>
+        machine.openshift.io/cluster-api-machine-type: <role> <2>
+        machine.openshift.io/cluster-api-machineset: <infrastructureID>-<role>-<region> <3>
+    spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          apiVersion: azureproviderconfig.openshift.io/v1beta1
+          credentialsSecret:
+            name: azure-cloud-credentials
+            namespace: openshift-machine-api
+          image:
+            offer: ""
+            publisher: ""
+            resourceID: /resourceGroups/<infrastructureID>-rg/providers/Microsoft.Compute/images/<infrastructureID>
+            sku: ""
+            version: ""
+          internalLoadBalancer: ""
+          kind: AzureMachineProviderSpec
+          location: centralus
+          managedIdentity: <infrastructureID>-identity <1>
+          metadata:
+            creationTimestamp: null
+          natRule: null
+          networkResourceGroup: ""
+          osDisk:
+            diskSizeGB: 128
+            managedDisk:
+              storageAccountType: Premium_LRS
+            osType: Linux
+          publicIP: false
+          publicLoadBalancer: ""
+          resourceGroup: <infrastructureID>-rg <1>
+          sshPrivateKey: ""
+          sshPublicKey: ""
+          subnet: <infrastructureID>-<role>-subnet <1> <2>
+          userDataSecret:
```

Comment:
For a new machineset, how do you need to change the subnet and userDataSecret parameter values?

Reply:
the `userDataSecret` should be the same than the other worker machineSets unless you create yourself a custom one